var/spack/repos/builtin/packages/python-venv/package.py

```diff
@@ -93,6 +93,10 @@ def setup_dependent_run_environment(self, env, dependent_spec):
             path = os.path.join(dependent_spec.prefix, directory)
             if os.path.isdir(path):
                 env.prepend_path("PYTHONPATH", path)
+        env.prepend_path("PATH", self.bindir)
+
+    def setup_dependent_build_environment(self, env, dependent_spec):
+        self.setup_dependent_run_environment(env, dependent_spec)
```

Comment:
We execute `setup_dependent_run_environment` for build/link/run dependencies, so I don't think this is necessary - did you see different behavior?

Or at least, I think so from reading through https://github.com/spack/spack/pull/35737 - can you confirm that @haampie?

Reply:
Ahhh that's the PR I was looking for. 

No I did not notice a difference in behavior, and was confused by that, so I added the quick one liner and figured it'd be easy to take out if I was wrong.

Reply:
Please delete those lines. The run environment only makes sense post install.

Reply:
> The run environment only makes sense post build. 

@haampie while I agree these lines should be removed for the functional reason I gave above, if you think this, then why is `setup_dependent_run_environment` run for build environment contexts?

Reply:
Build dependencies are used "at runtime"

Reply:
You mean: they are run during the build

A couple follow up questions:

* I assume we agree that `setup_dependent_build_environment` is for applying environment modifications that are only relevant at build time?
* Do you assume all runtime dependent environment modifications should be run at build time?

From reading through all related docstrings, I don't see anywhere this is specified, but this is not unique to these functions (see https://github.com/spack/spack/issues/3768: we don't have strong docs on what functionally occurs for the various dependency types).